_posts/2019-04-10-lecture-23.md

```diff
@@ -0,0 +1,401 @@
+## Inference in the Dirichlet Process <d-cite key="ishwaran2001gibbs">Gibbs Samplers</d-cite>, <d-cite key="hopper2015dpnotes">Dirichlet Process Notes</d-cite>
+
+Given a Dirichlet process mixture model and a dataset that presumably has been generated from the model, inference is the process of determining the component of the mixture model from which each of the data point came from. Let us consider a Dirichlet process mixture model: 
+
+
+$$\begin{aligned} 
+G := \sum_{k=1}^{\infty} \pi_{k} \delta_{\theta_{k}} & \sim \operatorname{DP}(\alpha, H) \\
+\phi_{n} & \sim G \\
+x_{n} & \sim f\left(\phi_{n}\right)
+\end{aligned}$$
+
+where the data point $x_{n}$ is generated from the component $\phi_{n}$ of the mixture model $G$. We want to determine the component $\phi_{n}$ for each data point $x_{n}$. 
+
+### Collapsed Gibbs Sampler
+The collapsed Gibbs Sampler integrates out $G$ to get a Chinese restaurant process (CRP). Since, samples in a CRP are exchangeable, we can always rearrange the ordering of data points so that any sampled data point $x_{n}$ is the last one. Let, $z_{n}$ be the cluster allocation of the $x_{n}$ and $K$ be the total number of instantiated clusters, then the probability of $x_{n}$ belonging to the $k^{th}$ cluster is given by:
+
+Note that, there is always a non-zero probability of the data point $x_{n}$ not being associated to any of the $K$ existing clusters and leads to the instantiation of a new cluster. 
+
+However, there are a few problems with this approach:
+
+- Updating only one data point at a time makes the algorithm infeasible for large datasets. 
+- In the case of two true clusters getting merged into a single cluster in the beginning (possibly due to bad initialization), it is unlikely that a single data point will break out and form a different cluster.
+- Getting to the true distribution involves going through low probability states and mixing can often be very slow.
+- Integrating out parameter for new features is difficult if the likelihood is not conjugate and requires approximation of the integration. 
+
+### Blocked Gibbs Sampler
+In the Blocked Gibbs Sampler, we instantiate $G$ instead of integrating it out. Although $G$ is infinite-dimensional, we can simply approximate it with a truncated stick-breaking process:
+
+$$\begin{aligned}
+G^{K} : &= \sum_{k=1}^{K} \pi_{k} \delta_{\theta_{k}} \\
+\pi_{k} &=b_{k} \prod_{j=1}^{k-1}\left(1-b_{j}\right) \\
+b_{k} & \sim \operatorname{Beta}(1, \alpha) \text { for } k=1, \ldots, K-1 \\
+b_{K} &= 1 
+\end{aligned}$$
+
+Now, for any $x_{n}$, its assocation probability to the $k^{th}$ cluster can be computed as:
+
+$$
+p(z_{n}|\text{rest}) \propto \pi_{k} f(w_{n}|\theta_{k})
+$$
+
+To estimate $\pi_{k}$ we follow the stick-breaking process which can also be thought of as a sequence of binary decisions. For example, we select $z_{n}=1$ with probability $b_{1}$. If $z_{n} \neq 1$, then we select $z_{n}=2$ with probability $b_{2}$ and so on. Formally, 
```

Comment:
This \neq does not appear to be rendering correctly. I also tried \not= and noticed the same result: slash appears several character spaces before the = symbol. This issue is present in the notes for lectures 4,9 and 12. The issue is repeatable in another browser. 
https://github.com/sailinglab/pgm-spring-2019/blame/master/_posts/2019-01-28-lecture-04.md#L290
https://github.com/sailinglab/pgm-spring-2019/blame/master/_posts/2019-02-13-lecture-09.md#L305
https://github.com/sailinglab/pgm-spring-2019/blame/master/_posts/2019-02-25-lecture-12.md#L319

Reply:
This is known KaTeX issue https://github.com/KaTeX/KaTeX/issues/1842
I've circumvented the issue by using =\not\

Reply:
Updating to KaTeX version "0.10.1" in the _config.yml file fixes this issue.